## E1701 (not-async-context-manager)

### :x: Problematic code:

```python
class ContextManager:
    def __enter__(self):
        pass

    def __exit__(self, *exc):
        pass


async def foo():
    async with ContextManager():
        pass
```

### :heavy_check_mark: Correct code:

```python
class AsyncContextManager:
    def __aenter__(self):
        pass

    def __aexit__(self, *exc):
        pass


async def foo():
    async with AsyncContextManager():
        pass
```

### Rationale:

Occurs when an async context manager is used with an object that does not
implement the async context management protocol, as an async context manager
does not implement `__aenter__` and `__aexit__`. If you are encountering this
error, please contact a TA for further help.

### Related resources:

- [Testcases](https://github.com/PyCQA/pylint/blob/master/tests/functional/n/not_async_context_manager.py)
- [Issue Tracker](https://github.com/PyCQA/pylint/issues?q=is%3Aissue+%22not-async-context-manager%22+OR+%22E1701%22)
